common/src/jni/main/cpp/org_conscrypt_NativeCrypto.cpp

```diff
@@ -1663,127 +1663,10 @@ static bool setBlocking(int fd, bool blocking) {
- * suppose there are not many other ways to do this on a Linux system (modulo
- * isomorphism).
- */
-#define MUTEX_TYPE pthread_mutex_t
-#define MUTEX_SETUP(x) pthread_mutex_init(&(x), nullptr)
-#define MUTEX_CLEANUP(x) pthread_mutex_destroy(&(x))
-#define MUTEX_LOCK(x) pthread_mutex_lock(&(x))
-#define MUTEX_UNLOCK(x) pthread_mutex_unlock(&(x))
-#define THREAD_ID pthread_self()
 #define THROW_SSLEXCEPTION (-2)
 #define THROW_SOCKETTIMEOUTEXCEPTION (-3)
 #define THROWN_EXCEPTION (-4)
 
-static MUTEX_TYPE* mutex_buf = nullptr;
-
-static void locking_function(int mode, int n, const char*, int) {
-    if (mode & CRYPTO_LOCK) {
-        MUTEX_LOCK(mutex_buf[n]);
-    } else {
-        MUTEX_UNLOCK(mutex_buf[n]);
-    }
-}
-
-/*
- * Wrapper for pthread_mutex_t to assist in unlocking in all paths.
- */
-class UniqueMutex {
-public:
-    explicit UniqueMutex(pthread_mutex_t* mutex) : mutex_(mutex) {
-        int err = pthread_mutex_lock(mutex_);
-        if (err != 0) {
-            ALOGE("failure obtaining mutex in %s: %d", __func__, err);
-            abort();
-        }
-        owns_ = true;
-    }
-
-    void unlock() {
-        if (owns_) {
-            owns_ = false;
-            int err = pthread_mutex_unlock(mutex_);
-            if (err != 0) {
-                ALOGE("failure releasing mutex in %s: %d", __func__, err);
-                abort();
-            }
-        }
-    }
-
-    ~UniqueMutex() {
-        unlock();
-    }
-
-private:
-    pthread_mutex_t* const mutex_;
-    bool owns_;
-};
-
-static void threadid_callback(CRYPTO_THREADID *threadid) {
-#if defined(__APPLE__)
-    uint64_t owner;
-    int rc = pthread_threadid_np(nullptr, &owner);  // Requires Mac OS 10.6
-    if (rc == 0) {
-        CRYPTO_THREADID_set_numeric(threadid, owner);
-    } else {
-        ALOGE("Error calling pthread_threadid_np");
-    }
-#else
-    // bionic exposes gettid(), but glibc doesn't
-    CRYPTO_THREADID_set_numeric(threadid, syscall(__NR_gettid));
-#endif
-}
-
-int THREAD_setup(void) {
-    mutex_buf = new MUTEX_TYPE[CRYPTO_num_locks()];
-    if (!mutex_buf) {
-        return 0;
-    }
-
-    for (int i = 0; i < CRYPTO_num_locks(); ++i) {
-        MUTEX_SETUP(mutex_buf[i]);
-    }
-
-    CRYPTO_THREADID_set_callback(threadid_callback);
-    CRYPTO_set_locking_callback(locking_function);
-
-    return 1;
-}
-
-int THREAD_cleanup(void) {
-    if (!mutex_buf) {
-        return 0;
-    }
-
-    CRYPTO_THREADID_set_callback(nullptr);
-    CRYPTO_set_locking_callback(nullptr);
-
-    for (int i = 0; i < CRYPTO_num_locks( ); i++) {
-        MUTEX_CLEANUP(mutex_buf[i]);
-    }
-
-    free(mutex_buf);
-    mutex_buf = nullptr;
-
-    return 1;
-}
-
-/**
- * Initialization phase for every OpenSSL job: Loads the Error strings, the
- * crypto algorithms and reset the OpenSSL library
- */
-static void NativeCrypto_clinit(JNIEnv*, jclass)
-{
-    SSL_load_error_strings();
-    ERR_load_crypto_strings();
-    SSL_library_init();
```

Comment:
This (`SSL_library_init`, which just calls `CRYPTO_library_init`) isn't quiiiite a no-op in an `BORINGSSL_NO_STATIC_INITIALIZER` build. That said, Chromium is the only thing that builds in that way because it's allergic to static initializers, so it's probably fine to omit it? Your call.